### PR TITLE
Simple date format each time new object creation removed

### PR DESCRIPTION
**Pull request by khanimteyaz**
## What is the purpose of the change
AcceLogFilter message creation and last and current check was performing by each time simple date formate object creation. In this version of code I have use the ThreadLocal for each thread wise to create simple date format object instead of each call

https://github.com/apache/incubator-dubbo/issues/3026

In this PR 

1.  I have moved the access log creation and writing to file into a single thread and reusing a single instance of **SimpleDateFormat**
2. Refactored code to separate our and group related task in separate method and have enhance the readability by using 
         -  Method naming 
         - Reducing big methods to small.

This PR is a resurrection of old 3027 (https://github.com/apache/incubator-dubbo/pull/3027) PR

Sorry for the inconvenience 
## Brief changelog

XXXXX

## Verifying this change

Runnig UT and verifying generated access log.

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue] (https://github.com/apache/incubator-dubbo/issues/3026) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by khanimteyaz** [comment]
@carryxyh I have raised the PR again for the issue #3026  and have closed the old #3027. 

As suggested I have moved the log message creating and writing to single threaded model so that single instance of **SimpleDateFormat** can be used.

**Comment by carryxyh** [comment]
Hi, @khanimteyaz 
Nice work. 
Your approach is completely in line with my vision. I will review it.

**Comment by khanimteyaz** [comment]
> Hi, @khanimteyaz
> Nice work.
> Your approach is completely in line with my vision. I will review it.

@carryxyh thanks.

File: dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/filter/AccessLogFilter.java [hunk 1]
```diff
@@ -67,139 +67,147 @@
...
+                writer.write("\r\n");
+            }
+            writer.flush();
+        } finally {
+            writer.close();
+        }
+    }
+
+    private AccessLogData buildAccessLogData(Invoker<?> invoker, Invocation inv) {
+        RpcContext context = RpcContext.getContext();
```
[the remark this answers is not in the document]
Comment (khanimteyaz) [reply dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/filter/AccessLogFilter.java]: done.

**Comment by khanimteyaz** [comment]
@carryxyh  It seems my rebase has introduce lots of commits of others, so raising another PR.

**Pull request closed without merging**